HW1-KZaballa.py

```diff
@@ -0,0 +1,181 @@
+# 25%      5.100000    2.800000    1.600000    0.300000
+# 50%      5.800000    3.000000    4.400000    1.300000
+# 75%      6.400000    3.300000    5.100000    1.800000
+# max      7.900000    4.400000    6.900000    2.500000
+
+
+# Plots:
+#   1)Scatter plot
+# 		add column headings to have something to anchor on when graphing
+iris_df.columns = [
+    "Sepal Length",
+    "Sepal Width",
+    "Petal Length",
+    "Petal Width",
+    "Class",
+]
+iris_df.head
+# 		plot here
+fig = px.scatter(
+    iris_df, x="Sepal Length", y="Sepal Width", color="Class", title="Iris Test Plot"
+)
+fig.show()
+
+#   2) Histogram
+# 		group by class and have a different color for each class
+fig0 = px.histogram(iris_df, x="Class", color="Class")
+fig0.show()
+
+#   3) Line graph
+# 		group by class and have a different color for each class
+#       This looks messy because it was a line graph that had to go
+#       through each point in the scatterplot. I will have to refine this
+#       later to just reflect a trendline.
+fig1 = px.line(iris_df, x="Petal Length", y="Petal Width", color="Class")
+fig1.show()
+
+results = px.get_trendline_results(fig1)
+print(results)
+
+#   4) Pie chart
+#
+fig3 = px.pie(
+    iris_df,
+    names="Class",
+    values="Petal Width",
+    color="Class",
+    title="Petal Width by class in different colors",
+)
+fig3.show()
+
+#   5) histogram, scatter and rug
+#   double graphs
+fig4 = px.scatter(
+    iris_df,
+    x="Sepal Length",
+    y="Sepal Width",
+    marginal_x="histogram",
+    marginal_y="rug",
+    color="Class",
+)
+fig4.show()
+
+#   6) Different scatter-line plot styles
+#
+iris_df.head
+fig5 = go.Figure()
+fig5.add_trace(
+    go.Scatter(
+        x=iris_df["Sepal Length"],
+        y=iris_df["Sepal Width"],
+        mode="lines+markers",
+        name="lines+markers",
+    )
+)
+fig5.add_trace(
+    go.Scatter(
+        x=iris_df["Sepal Length"],
+        y=iris_df["Petal Length"],
+        mode="lines+markers",
+        name="lines+markers",
+    )
+)
+fig5.add_trace(
+    go.Scatter(
+        x=iris_df["Sepal Length"],
+        y=iris_df["Petal Width"],
+        mode="markers",
+        name="markers",
+    )
+)
+fig5.show()
+
+#   7) Violin, box, with main scatter plot
+fig6 = px.scatter(
+    iris_df,
+    x=iris_df["Sepal Length"],
+    y=iris_df["Sepal Width"],
+    color="Class",
+    marginal_x="box",
+    marginal_y="violin",
+    title="Click on the legend items!",
+)
+fig6.show()
+
+#   Normalize and transform
+# Normalizer does not accept qualitative data
+X = iris_df[["Sepal Length", "Sepal Width", "Petal Length", "Petal Width"]]
+X_trans = Normalizer().transform(X)
+# set norm to mean of sepal length and transform
+# it by that
+#
+# Is my logic correct here? I am thinking in
+# order to truly normalize something, each max or mean
+# or whatever function has to be applied to each column
+# transformer = Normalizer(norm=max(X["Sepal Length"]),

```

Comment:
You should look at the documentation if you want to change the normal behavior of the normalizer. It was just a silly function I wanted you guys to use,.. not really useful really.